Snowflake.Data/Client/SnowflakeDbParameter.cs

```diff
@@ -34,7 +36,28 @@ public SnowflakeDbParameter(int ParameterIndex, SFDataType SFDataType)
             this.SFDataType = SFDataType;
         }
 
-        public override DbType DbType { get; set; }
+        public override DbType DbType
+        {
+            get
+            {
+                if (_dbType != DbType.AnsiString || Value == null || Value is DBNull)

```

Comment:
what happens when _dbType and Value is null?

Reply:
I tried to test this but only `Value` can be set to null. The `_dbType` can't be set to null because of this error: `Cannot convert null to 'type' because it is a non-nullable value type`

It looks like `_dbType` will always default to `DbType.AnsiString` and the code would return the default `DbType`